src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -167,54 +167,22 @@ internal SequencePosition Seek(SequencePosition start, SequencePosition end, lon
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static SequencePosition SeekMultiSegment(IMemoryList<byte> start, int startIndex, IMemoryList<byte> end, int endPosition, long count, bool checkEndReachable)
+        private static SequencePosition SeekMultiSegment(IMemoryList<byte> start, int startIndex, IMemoryList<byte> end, int endPosition, long count)
         {
-            SequencePosition result = default;
-            bool foundResult = false;
-            IMemoryList<byte> current = start;
-            int currentIndex = startIndex;
-
-            while (current != null)
+            var memoryList = start.GetNext(startIndex + count, out int localOffset);
+            if (memoryList == null ||
```

Comment:
This is a private method, should this be a Debug.Assert instead?

Reply:
It's called by a public method and this is the only check.

Reply:
`GetNext` returns null when offset was outside of available list so this is not assert it's more of bounds checking logic.